File: ios/Packages/FeatureOrders/Sources/FeatureOrders/Domain/Order.swift
import Foundation

public struct Order: Identifiable, Equatable {
    public let id: String
    public let title: String
    public let amount: Double
    public let status: OrderStatus

    public init(id: String, title: String, amount: Double, status: OrderStatus) {
        self.id = id
        self.title = title
        self.amount = amount
        self.status = status
    }
}

public enum OrderStatus: String, Equatable {
    case pending, processing, completed, cancelled
}

File: flutter/feature_orders/lib/src/domain/order.dart
/// Domain model representing an order.
class Order {
  final String id;
  final String title;
  final double amount;
  final OrderStatus status;

  const Order({
    required this.id,
    required this.title,
    required this.amount,
    required this.status,
  });
}

enum OrderStatus { pending, processing, completed, cancelled }
